<launch>
	<arg name="sub_joint_state_topic_name"     default="/joint_states" />
	<arg name="pub_base_twist_topic_name"      default="/command_velocity" />
	<arg name="pub_trajectory_topic_name"      default="/controller/command" />
	<arg name="pub_hand_trajectory_topic_name" default="/hand_controller/command" />
	
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<node name="ai_agent_teleop_key" pkg="sigverse_ai_agent_teleop_key" type="ai_agent_teleop_key" output="screen" launch-prefix="gnome-terminal -t 'AI Agent teleop key' -e">

		<param name="sub_joint_state_topic_name"     type="string" value="$(arg sub_joint_state_topic_name)" />
		<param name="pub_base_twist_topic_name"      type="string" value="$(arg pub_base_twist_topic_name)" />
		<param name="pub_trajectory_topic_name"      type="string" value="$(arg pub_trajectory_topic_name)" />
		<param name="pub_hand_trajectory_topic_name" type="string" value="$(arg pub_hand_trajectory_topic_name)" />
	</node>

	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
	</group>
	
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" > 
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>
	
</launch>

